base/set.jl

```diff
@@ -591,15 +591,14 @@ Set([0, 2, 3])
 """
 replace!(A, old_new::Pair...; count::Integer=typemax(Int)) = _replace!(A, eltype(A), count, old_new)
 
-# we use this wrapper because using directly eltype(A) as the type
-# parameter below for Some degrades performance
 function _replace!(A, ::Type{K}, count::Integer, old_new::Tuple{Vararg{Pair}}) where K
```

Comment:
seems like we can drop `K` now too?

Reply:
Probably, and I also wanted to check if with this new version we could get rid of this intermediate function altogether without performance hit, but was lacking time. I will come to it eventually.